2018-komp-ling/projects/serikov.md

```diff
@@ -0,0 +1,75 @@
+# Compling project proposal
+
+## Abstract
+RE: extracting phonological rules from neural networks paper: Jennifer Rodd (1997) "Recurrent Neural-Network Learning of Phonological Regularities in Turkish". CoNLL97: Computational Natural Language Learning, http://www.aclweb.org/anthology/W97-1012
+
+## Introduction
+**TO-DO** the background of the problem to be solved or topic to be studied, with references for the low-background reader.
+
+## Proposed goals
+
+### MVP
+* Re-implement the paper
+* Release the code and documentation showing how to run it
+* Extract the graphs that are contained in the original paper
+
+### EP
+Run the NNs explored in the original paper with other languages in the Turkic family to see if their networks split the same way (other Turkic languages have different vowel harmony systems, and Uzbek does not have vowel harmony at all). 
+
+### HAP
+Apply the received knowledge of computational phonology and Turkic family languages phonology in particular to solve the task of phonetic embeddings alignment without having big parallel corpus.
+
+## Requirements
+
+**NB!** Each sub-goal is accompanied with the documentation composing and sources publishing where needed.
+
+**NB!** Goals may be changed on the run if necessary (e.g. smth unexpected discovered). All the possible changes should be discussed with mentors.
+
+### MVP requirements
+#### Skills required
+* Brief knowledge of actual NN creating techniques.
+* Data preprocessing skills
+* Vizualization skill.
+
+#### Sub-goals
+* 1 week| Reproduce the dataset used in original paper
+* 1/2 week| Reproduce the NNs used in original paper
+* 1/2 week| Reproduce the training process used in original paper
+* 1/2 week| Reproduce the NNs analysis described in original paper
+* 1/4 week| Compare the reproduced and original results
+* 1/4 week| Draft the results report
+* 1/4 week| Discuss the results with mentors
+* 1/4 week| Report the results of the MVP stage online (e.g. repo readme)
+
+### EP requirements
+
+#### Sub-goals
+* 1 week| Collect the data to repeat the research on different languages data
```

Comment:
See above.

Reply:
Pretty much any of the Turkic languages, you can do something like:

```
$ cat apertium-tur//apertium-tur.tur.lexc  |\
 grep -v '^!' |\
 grep '[^<> ]\+:[^<> ]\+ \(N[^PU]\|V\)[^ ]\+ ;' | cut -f1 -d':' |\
 sort -Ru | head -1000
```

From [apertium-tur](github.com/apertium/apertium-tur). I'm happy to generate the data for you.